src/Stryker.Core/Stryker.Core/Mutators/LinqMutator.cs

```diff
@@ -0,0 +1,81 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Stryker.Core.Enumerations;
+using Stryker.Core.Mutants;
+
+namespace Stryker.Core.Mutators
+{
+    /// <summary>
+    ///     Mutator Implementation for LINQ Mutations
+    /// </summary>
+    public class LinqMutator : Mutator<IdentifierNameSyntax>, IMutator
+    {
+        #region Private Properties
+
+        /// <summary>
+        ///     Dictionary which maps original linq expressions to the target mutation
+        /// </summary>
+        private Dictionary<LinqExpression, LinqExpression> _kindsToMutate { get; }
+
+        #endregion
+
+        #region Constructor
+
+        /// <summary>
+        ///     Constructor for the <see cref="LinqMutator"/>
+        /// </summary>
+        public LinqMutator()
+        {
+            _kindsToMutate = new Dictionary<LinqExpression, LinqExpression>
+            {
+                { LinqExpression.FirstOrDefault, LinqExpression.SingleOrDefault },
+                { LinqExpression.SingleOrDefault, LinqExpression.FirstOrDefault },
+                { LinqExpression.First, LinqExpression.Last },
+                { LinqExpression.Last, LinqExpression.First },
+                { LinqExpression.All, LinqExpression.Any },
+                { LinqExpression.Any, LinqExpression.All },
+                { LinqExpression.Skip, LinqExpression.Take },
+                { LinqExpression.Take, LinqExpression.Skip },
+                { LinqExpression.SkipWhile, LinqExpression.TakeWhile },
+                { LinqExpression.TakeWhile, LinqExpression.SkipWhile },
+                { LinqExpression.Min, LinqExpression.Max },
+                { LinqExpression.Max, LinqExpression.Min },
+                { LinqExpression.Sum, LinqExpression.Count },
+                { LinqExpression.Count, LinqExpression.Sum }
+            };
+        }
+
+        #endregion
+
+        #region Mutation Overrides
+
+        /// <summary>
+        ///     Apply mutations to an <see cref="IdentifierNameSyntax"/>
+        /// </summary>
+        /// <param name="node"></param>
+        /// <returns></returns>
+        public override IEnumerable<Mutation> ApplyMutations(IdentifierNameSyntax node)
+        {
+            if (Enum.TryParse(node.Identifier.ValueText, out LinqExpression expression) &&
+                _kindsToMutate.TryGetValue(expression, out LinqExpression replacementExpression))
```

Comment:
Can this ever be false? I think this can be placed inside the ifstatement and maybe doesn't have to be a Try method?

Reply:
It can indeed throw an exception, whenever some NameSyntax equals to None which is no key in the enum.
I've just added the None, since I overlooked the Removal of some Linq Queries before.